perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java

```diff
@@ -13,10 +25,180 @@
+
+	private final static RowMapper<ConsentHub> CONSENT_HUB_MAPPER = (rs, rowNum) -> {
+		return new ConsentHub();
+	};
+
+	// Consent mapper
+	private final static RowMapper<Consent> CONSENT_MAPPER = (rs, rowNum) -> {
+		Consent c = new Consent();
+		c.setId(rs.getInt("consents_id"));
+		c.setUserId(rs.getInt("consents_user_id"));
+		c.setConsentHub(CONSENT_HUB_MAPPER.mapRow(rs, rowNum));
+		c.setStatus(ConsentStatus.valueOf(rs.getString("consents_status")));
+		c.setCreatedAt(rs.getString("consents_created_at"));
+		c.setCreatedBy(rs.getString("consents_created_by"));
+		c.setModifiedAt(rs.getString("consents_modified_at"));
+		c.setModifiedBy(rs.getString("consents_modified_by"));
+		if(rs.getInt("consents_modified_by_uid") == 0) c.setModifiedByUid(null);
+		else c.setModifiedByUid(rs.getInt("consents_modified_by_uid"));
+		if(rs.getInt("consents_created_by_uid") == 0) c.setCreatedByUid(null);
+		else c.setCreatedByUid(rs.getInt("consents_created_by_uid"));
+		return c;
+	};
+
 	private static JdbcPerunTemplate jdbc;
 
 	public ConsentsManagerImpl(DataSource perunPool) {
 		jdbc = new JdbcPerunTemplate(perunPool);
 		jdbc.setQueryTimeout(BeansUtils.getCoreConfig().getQueryTimeout());
 	}
+
+	@Override
+	public List<Consent> getAllConsents(PerunSession sess) {
+		try {
+			List<Consent> consents = jdbc.query("select " + consentMappingSelectQuery + ", " + consentHubMappingSelectQuery +
+				" from consents left join consent_hubs on consents.consent_hub_id=consent_hubs.id ", CONSENT_MAPPER);
+			for (Consent consent : consents) {
+				if (consent != null) {
+					consent.setAttributes(getAttrDefsForConsent(sess, consent.getId()));
+				}
+			}
+			return consents;
+		} catch (EmptyResultDataAccessException ex) {
+			return new ArrayList<>();
+		} catch (RuntimeException err) {
+			throw new InternalErrorException(err);
+		}
+	}
+
+	@Override
+	public List<Consent> getConsentsForConsentHub(PerunSession sess, int id, ConsentStatus status) {
+		try {
+			List<Consent> consents = jdbc.query("select " + consentMappingSelectQuery + ", " + consentHubMappingSelectQuery +
+				" from consents left join consent_hubs on consents.consent_hub_id=consent_hubs.id " + " where consents.consent_hub_id=? and consents.status=? ", CONSENT_MAPPER, id, status);
+			for (Consent consent : consents) {
+				if (consent != null) {
+					consent.setAttributes(getAttrDefsForConsent(sess, consent.getId()));
+				}
+			}
+			return consents;
+		} catch (EmptyResultDataAccessException ex) {
+			return new ArrayList<>();
+		} catch (RuntimeException err) {
+			throw new InternalErrorException(err);
+		}
+	}
+
+	@Override
+	public List<Consent> getConsentsForConsentHub(PerunSession sess, int id) {
+		try {
+			List<Consent> consents = jdbc.query("select " + consentMappingSelectQuery + ", " + consentHubMappingSelectQuery +
+				" from consents left join consent_hubs on consents.consent_hub_id=consent_hubs.id " + " where consents.consent_hub_id=? ", CONSENT_MAPPER, id);
```

Comment:
This could be optimized - we don't need to join tables; instead, we can get the consent hub by id and then set it to every consent in the following for-cycle. Then we will also get the facilities list to the consent hub object. Usually, the performance wouldn't be so important, but this method will be used during data propagation so it should be as fast as possible.